test/geospatial_test.dart

```diff
@@ -0,0 +1,335 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'dart:async';
+import 'dart:math';
+
+import 'package:realm_common/realm_common.dart';
+import 'package:test/test.dart' hide test, throws;
+
+import '../lib/realm.dart';
+import 'test.dart';
+
+part 'geospatial_test.g.dart';
+
+@RealmModel(ObjectType.embeddedObject)
+class _Location {
+  final String type = 'Point';
+  List<double> coordinates = const [0, 0];
+
+  double get longitude => coordinates[0];
+  set longitude(double value) => coordinates[0] = value;
+
+  double get latitude => coordinates[1];
+  set latitude(double value) => coordinates[1] = value;
+}
+
+@RealmModel()
+class _Restaurant {
```

Comment:
I realized we don't have any tests for collections of geo points in .NET. Not sure if this is covered by other SDK tests, but might be a good idea to add some for dart at least.

Reply:
Isn't that just a test that we can store embedded objects in a list?

Reply:
Well, not just store - also run geo queries. While I do hope that's covered by Core tests, it never hurts to double check.

Reply:
Done